hr_holidays_compute_days/models/hr_holidays.py

```diff
@@ -1,101 +1,121 @@
 
-    @api.onchange('employee_id')
-    def _onchange_employee(self):
-        super(HrHolidays, self)._onchange_employee()
-        self._check_and_recompute_days()
+    @api.depends('date_to')
+    def _compute_date_to_full(self):
+        """Put day in employee's user timezone, or user timezone as fallback"""
+        for record in self.filtered('date_to'):
+            tz_name = record.employee_id.user_id.tz or record.env.user.tz
+            dt = fields.Datetime.from_string(record.date_to).replace(
+                tzinfo=tz.tzutc(),
+            ).astimezone(tz.gettz(tz_name)).date()
+            record.date_to_full = fields.Date.to_string(dt)
 
-    def _check_and_recompute_days(self):
-        date_from = self.date_from
-        date_to = self.date_to
-        if (date_to and date_from) and (date_from <= date_to):
-            if not self._check_date_helper(self.employee_id.id, date_from):
-                raise ValidationError(_("You cannot schedule the start date "
-                                        "on a public holiday or employee's "
-                                        "rest day"))
-            if not self._check_date_helper(self.employee_id.id, date_to):
-                raise ValidationError(_("You cannot schedule the end date "
-                                        "on a public holiday or employee's "
-                                        "rest day"))
-            self.number_of_days_temp = self._recompute_number_of_days()
+    def _inverse_date_from_full(self):
+        """Put start of the day in employee's user timezone, or user timezone
+        as fallback.
+        """
+        for record in self.filtered('date_from_full'):
+            tz_name = record.employee_id.user_id.tz or record.env.user.tz
+            dt = fields.Datetime.from_string(record.date_from_full).replace(
+                hour=0, minute=0, second=0, microsecond=0,
+                tzinfo=tz.gettz(tz_name),
+            ).astimezone(tz.tzutc())
+            record.date_from = fields.Datetime.to_string(dt)
 
-    @api.onchange('date_from')
-    def _onchange_date_from(self):
-        super(HrHolidays, self)._onchange_date_from()
-        employee_id = self.employee_id.id
-        if not self._check_date_helper(employee_id, self.date_from):
-            raise ValidationError(_("You cannot schedule the start date on "
-                                    "a public holiday or employee's rest day"))
-        if (self.date_to and self.date_from) \
-           and (self.date_from <= self.date_to):
-            self.number_of_days_temp = self._recompute_number_of_days()
+    def _inverse_date_to_full(self):
+        """Put end of the day in employee's user timezone, or user timezone
+        as fallback.
+        """
+        for record in self.filtered('date_to_full'):
+            tz_name = record.employee_id.user_id.tz or record.env.user.tz
+            dt = fields.Datetime.from_string(record.date_to_full).replace(
```

Comment:
This i think this we have to change, to allow user select, because on partial holidays, they can select a holiday at beginning of the day, let's say from 08 to 12. Even if i remove the calculation in holiday status to allow partial results, remove from holiday view the full day checkbox to select hours, i cannot select hours for date_to, the hours are changed to 23:59:59 every time...

Reply:
Yeah, I have noticed this also. I'm on it.

Reply:
This has been solved with latest code.